src/views/Timesheets/index.js

```diff
@@ -356,6 +397,31 @@ const Timesheets = (props) => {
       return shouldDisableDate;
     };
 
+    const clockIn = async() =>{
+
+      if(clockInOut === "Clock In"){
+        setClockInOut("Clock Out")
+        await jobs.clockIn(true);
+        let clockInDate = new Date();
+        handleDateChangeIn(clockInDate);
+
+      } 
+      if(clockInOut === "Clock Out") {
+          setClockInOut("Reset");
+          await jobs.clockIn(false);
+          let clockOutDate = new Date();
+          handleDateChangeOut(clockOutDate);
+          await jobs.deleteClockIn();
+      } 
+      if(clockInOut === "Reset") {
+          setClockInOut("Clock In");
+          setSelectedDateIn(null);
+          setSelectedDateOut(null);
+
```

Comment:
Everything looks good to me. There are just two things. The first one is the same as what Cameron pointed out. I was confused seeing that function was called clockIn even though it handled clocking in, out, and resetting. The second thing are the if statements. Did you mean to use "else if" for line 408 and 515? Without it, each if statement will process. It's not really a big deal at all since it very minuscule code haha.